streams/src/test/java/org/apache/kafka/streams/smoketest/SmokeTestDriver.java

```diff
@@ -143,14 +163,22 @@ public void run() {
         }
         Random rand = new Random();
 
-        int remaining = data.length;
+        int remaining = 0;
+        if (autoTerminate) {
+            remaining = data.length;
+        }
 
-        while (remaining > 0) {
-            int index = rand.nextInt(remaining);
+        while (running || remaining > 0) {
```

Comment:
The logic is a bit awkward to read, could we keep `while (running)` here and inside the while loop and inside the condition line 181 add

```
if (remaining ==0)   // we know autoTerminate == true for sure
    running = false
```

Then we do not need the line 137 above to immediately negate the running flag.